xbmc/guilib/GUIWindowManager.cpp

```diff
@@ -458,17 +458,17 @@ bool CGUIWindowManager::SendMessage(CGUIMessage& message)
   if (message.GetMessage()==GUI_MSG_NOTIFY_ALL)
   {
     CSingleLock lock(g_graphicsContext);
-    for (rDialog it = m_activeDialogs.rbegin(); it != m_activeDialogs.rend(); ++it)
+
+    for (const auto& window : m_activeDialogs)
```

Comment:
after a second look: I don't know why we force reverse order to call OnMessage() .. it doesn't matter if you ask me. 2  lines below we iterate over all windows without a strict order ;)